wpimath/src/main/java/edu/wpi/first/math/system/LinearSystem.java

```diff
@@ -196,6 +196,17 @@ public Matrix<Outputs, N1> calculateY(Matrix<States, N1> x, Matrix<Inputs, N1> c
     return m_C.times(x).plus(m_D.times(clampedU));
   }
 
+  /**
+   * Returns the LinearSystem with the number of outputs set to 1
+   *
+   * <p>This is used by state observers such as the Kalman Filter.
+   *
+   * @return the trimmed LinearSystem with outputs set to 1.
+   */
+  public LinearSystem<States, Inputs, N1> getTrimmedLinearSystem() {
+    return new LinearSystem<>(m_A, m_B, m_C.extractRowVector(0), m_D.extractRowVector(0));
```

Comment:
This is too specific. The user could slice any outputs in theory because we don't know what kind of system they've made. I'd suggest something like this:
```java
public LinearSystem<States, Inputs, ?> slice(int... outputIndices);
```
Idk how to compute the third generic argument in Java, but in C++ it's just `sizeof...(outputIndices)`.

Reply:
I have a not so elegant way to do this.  Since there are only N0...N20 (21) different possibilities we could just navigate them all based on a switch statement.  

I don't think there is a way to programmatically determine the third generic argument.

Reply:
That's probably what we'll have to do. I remember us hitting similar issues in the unscented Kalman filter because Java's type system is a potato.

Reply:
Okay I've updated with an attempt.  

Just a thought.  Could we instead just add a identifyPositionSystemPartial for use with KalmanFilters, etc.

Reply:
I'd rather take the casts than duplicate the API.

Reply:
Also, we're fine with complicating the internals in the name of getting a specific API shape. That's the MO of C++ in general w/r/t library dev.